### ISSUE TITLE
Hoodie app always eventually terminates with: "RangeError: Maximum call stack size exceeded"

### ISSUE THREAD

**Issue by quaxquax** [hoodiehq/faq#91]
At this point in early development this is not much of a concern to me, but it would be in production. 

It seems that whatever app I have running, after a while it crashes with the RangeError. Seen this happen consistently. A full error trace of such a crash is attached below. (This particular app is derived from the example app that the "hoodie new" command creates by default).

Will make some more tests to see if this is specific to my test apps based on this particular example.

Hoodie is version 0.6.3 (node v0.10.36, npm 2.7.0, platform: darwin)

My development platform is an older Macbook Pro OSX 10.9.5, CouchDB 1.3

Figure the first thing for me to try (if this is not app specific) will be to update my CouchDB. 

#################### Error trace #####################

Debug: hapi, internal, implementation, error 
    TypeError: Cannot call method 'code' of undefined
    at internals.addCorsAndBearerToken (/private/tmp/teststuff/node_modules/hoodie-server/lib/server/plugins/api/index.js:34:18)
    at /private/tmp/teststuff/node_modules/hoodie-server/node_modules/hapi/node_modules/h2o2/lib/index.js:139:52
    at finish (/private/tmp/teststuff/node_modules/hoodie-server/node_modules/wreck/lib/index.js:92:20)
    at wrapped (/private/tmp/teststuff/node_modules/hoodie-server/node_modules/wreck/node_modules/hoek/lib/index.js:813:20)
    at ClientRequest.onError (/private/tmp/teststuff/node_modules/hoodie-server/node_modules/wreck/lib/index.js:101:16)
    at ClientRequest.g (events.js:180:16)
    at ClientRequest.emit (events.js:95:17)
    at Socket.socketOnEnd [as onend] (http.js:1573:9)
    at Socket.g (events.js:180:16)
    at Socket.emit (events.js:117:20)
Error getting update from changes pool
{ [Error: socket hang up]
  code: 'ECONNRESET',
  domainEmitter: 
   { domain: 
      { domain: null,
        _events: [Object],
        _maxListeners: 10,
        members: [] },
     _events: { pipe: [Object], response: [Function], end: [Function] },
     _maxListeners: 10,
     readable: true,
     writable: true,
     method: 'GET',
     uri: 
      { protocol: 'http:',
        slashes: true,
        auth: '_hoodie:Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej/1RZeehFZIaeMSVPbQkywG/hJLzlwVynonwqUMShEajqncopg/n8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE/7hUmFzDEh+36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr+bRzn7dW4kgnQIL2LLuEYKs+B0bcga1/UJlkgPywVN6GQUumeCGVM2aZqY8em2doH/6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA==',
        host: '127.0.0.1:6015',
        port: '6015',
        hostname: '127.0.0.1',
        hash: null,
        search: '?since=7&feed=continuous&heartbeat=30000&include_docs=true',
        query: 'since=7&feed=continuous&heartbeat=30000&include_docs=true',
        pathname: '/user%2F789bh5f/_changes',
        path: '/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
        href: 'http://_hoodie:Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej%2F1RZeehFZIaeMSVPbQkywG%2FhJLzlwVynonwqUMShEajqncopg%2Fn8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE%2F7hUmFzDEh%2B36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr%2BbRzn7dW4kgnQIL2LLuEYKs%2BB0bcga1%2FUJlkgPywVN6GQUumeCGVM2aZqY8em2doH%2F6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA%3D%3D@127.0.0.1:6015/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true' },
     headers: 
      { accept: 'application/json',
        host: '127.0.0.1:6015',
        authorization: 'Basic X2hvb2RpZTpDbDBFaEVKbmNodTV1QmFvbUtmTHJYRktUT0hST2xld21wb09aeXROTTNWcWpPUzdHcE9QdmFxMGVqLzFSWmVlaEZaSWFlTVNWUGJRa3l3Ry9oSkx6bHdWeW5vbndxVU1TaEVhanFuY29wZy9uOHlmSXFOME1KY1NZaG4xREFiTFk3dmhhVXpEbWxFLzdoVW1GekRFaCszNkxaRlRQR3JKZ0d6U1VsV2Fub3dHVmhkemFkdU1PZ016aXVLV3R1QzVMZ3QxZVI0eUhjQUp1d3IrYlJ6bjdkVzRrZ25RSUwyTEx1RVlLcytCMGJjZ2ExL1VKbGtnUHl3Vk42R1FVdW1lQ0dWTTJhWnFZOGVtMmRvSC82UWNUdVdKMVp5aGJpR3Z6WGZ4Z3YzaEJncm5IUkJCU2tSNnIzUDFMMlBDbnFkeWVtOW1tMEtEVjF0N2ZPS3JOQ0t1dkE9PQ==' },
     encoding: 'utf-8',
     changes_query: 
      { since: 7,
        feed: 'continuous',
        heartbeat: 30000,
        include_docs: true },
     pool: {},
     dests: [],
     __isRequestRequest: true,
     _redirectsFollowed: 0,
     maxRedirects: 10,
     followRedirect: true,
     followAllRedirects: false,
     redirects: [],
     setHost: true,
     originalCookieHeader: undefined,
     _jar: undefined,
     port: '6015',
     host: '127.0.0.1',
     clientErrorHandler: [Function],
     _parserErrorHandler: [Function],
     path: '/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
     _user: '_hoodie',
     _pass: 'Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej/1RZeehFZIaeMSVPbQkywG/hJLzlwVynonwqUMShEajqncopg/n8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE/7hUmFzDEh+36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr+bRzn7dW4kgnQIL2LLuEYKs+B0bcga1/UJlkgPywVN6GQUumeCGVM2aZqY8em2doH/6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA==',
     _hasAuth: true,
     _sentAuth: true,
     httpModule: 
      { parsers: [Object],
        STATUS_CODES: [Object],
        IncomingMessage: [Object],
        OutgoingMessage: [Object],
        ServerResponse: [Object],
        Agent: [Object],
        globalAgent: [Object],
        ClientRequest: [Object],
        request: [Function],
        get: [Function],
        Server: [Object],
        createServer: [Function],
        _connectionListener: [Function: connectionListener],
        Client: [Function: deprecated],
        createClient: [Function: deprecated] },
     agentClass: { [Function: Agent] super_: [Object], defaultMaxSockets: 5 },
     agent: 
      { domain: null,
        _events: [Object],
        _maxListeners: 10,
        options: {},
        requests: {},
        sockets: [Object],
        maxSockets: 5,
        createConnection: [Function] },
     _started: true,
     href: 'http://_hoodie:Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej%2F1RZeehFZIaeMSVPbQkywG%2FhJLzlwVynonwqUMShEajqncopg%2Fn8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE%2F7hUmFzDEh%2B36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr%2BbRzn7dW4kgnQIL2LLuEYKs%2BB0bcga1%2FUJlkgPywVN6GQUumeCGVM2aZqY8em2doH%2F6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA%3D%3D@127.0.0.1:6015/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
     req: 
      { domain: [Object],
        _events: [Object],
        _maxListeners: 10,
        output: [],
        outputEncodings: [],
        writable: true,
        _last: true,
        chunkedEncoding: false,
        shouldKeepAlive: true,
        useChunkedEncodingByDefault: false,
        sendDate: false,
        _headerSent: true,
        _header: 'GET /user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true HTTP/1.1\r\naccept: application/json\r\nhost: 127.0.0.1:6015\r\nauthorization: Basic X2hvb2RpZTpDbDBFaEVKbmNodTV1QmFvbUtmTHJYRktUT0hST2xld21wb09aeXROTTNWcWpPUzdHcE9QdmFxMGVqLzFSWmVlaEZaSWFlTVNWUGJRa3l3Ry9oSkx6bHdWeW5vbndxVU1TaEVhanFuY29wZy9uOHlmSXFOME1KY1NZaG4xREFiTFk3dmhhVXpEbWxFLzdoVW1GekRFaCszNkxaRlRQR3JKZ0d6U1VsV2Fub3dHVmhkemFkdU1PZ016aXVLV3R1QzVMZ3QxZVI0eUhjQUp1d3IrYlJ6bjdkVzRrZ25RSUwyTEx1RVlLcytCMGJjZ2ExL1VKbGtnUHl3Vk42R1FVdW1lQ0dWTTJhWnFZOGVtMmRvSC82UWNUdVdKMVp5aGJpR3Z6WGZ4Z3YzaEJncm5IUkJCU2tSNnIzUDFMMlBDbnFkeWVtOW1tMEtEVjF0N2ZPS3JOQ0t1dkE9PQ==\r\nConnection: keep-alive\r\n\r\n',
        _hasBody: true,
        _trailer: '',
        finished: true,
        _hangupClose: false,
        socket: [Object],
        connection: [Object],
        agent: [Object],
        socketPath: undefined,
        method: 'GET',
        path: '/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
        _headers: [Object],
        _headerNames: [Object],
        parser: [Object] },
     ntick: true },
  domain: 
   { domain: null,
     _events: { error: [Function] },
     _maxListeners: 10,
     members: [] },
  domainThrown: false }

events.js:53
EventEmitter.prototype.emit = function emit(type) {

**Comment by janl** [comment]
Which version of hoodie-server are you on?

Best
Jan
--

> On 23 Mar 2015, at 07:15, quaxquax <notifications@github.com> wrote:
> 
> At this point in early development this is not much of concern, but it would be in production.
> 
> No matter what app I have running, after a while it crashes with the RangeError. A full error trace of such a crash is attached below.
> 
> Hoodie is version 0.6.3 (node v0.10.36, npm 2.7.0, platform: darwin)
> 
> My development platform is an older Macbook Pro OSX 10.9.5, CouchDB 1.3
> 
> Figure the first thing for me to try is to update my CouchDB.
> 
> #################### Error trace #####################
> 
> Debug: hapi, internal, implementation, error 
> TypeError: Cannot call method 'code' of undefined
> at internals.addCorsAndBearerToken (/private/tmp/teststuff/node_modules/hoodie-server/lib/server/plugins/api/index.js:34:18)
> at /private/tmp/teststuff/node_modules/hoodie-server/node_modules/hapi/node_modules/h2o2/lib/index.js:139:52
> at finish (/private/tmp/teststuff/node_modules/hoodie-server/node_modules/wreck/lib/index.js:92:20)
> at wrapped (/private/tmp/teststuff/node_modules/hoodie-server/node_modules/wreck/node_modules/hoek/lib/index.js:813:20)
> at ClientRequest.onError (/private/tmp/teststuff/node_modules/hoodie-server/node_modules/wreck/lib/index.js:101:16)
> at ClientRequest.g (events.js:180:16)
> at ClientRequest.emit (events.js:95:17)
> at Socket.socketOnEnd as onend
> at Socket.g (events.js:180:16)
> at Socket.emit (events.js:117:20)
> Error getting update from changes pool
> { [Error: socket hang up]
> code: 'ECONNRESET',
> domainEmitter: 
> { domain: 
> { domain: null,
> events: [Object],
> _maxListeners: 10,
> members: [] },
> _events: { pipe: [Object], response: [Function], end: [Function] },
> _maxListeners: 10,
> readable: true,
> writable: true,
> method: 'GET',
> uri: 
> { protocol: 'http:',
> slashes: true,
> auth: '_hoodie:Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej/1RZeehFZIaeMSVPbQkywG/hJLzlwVynonwqUMShEajqncopg/n8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE/7hUmFzDEh+36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr+bRzn7dW4kgnQIL2LLuEYKs+B0bcga1/UJlkgPywVN6GQUumeCGVM2aZqY8em2doH/6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA==',
> host: '127.0.0.1:6015',
> port: '6015',
> hostname: '127.0.0.1',
> hash: null,
> search: '?since=7&feed=continuous&heartbeat=30000&include_docs=true',
> query: 'since=7&feed=continuous&heartbeat=30000&include_docs=true',
> pathname: '/user%2F789bh5f/_changes',
> path: '/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
> href: 'http://_hoodie:Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej%2F1RZeehFZIaeMSVPbQkywG%2FhJLzlwVynonwqUMShEajqncopg%2Fn8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE%2F7hUmFzDEh%2B36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr%2BbRzn7dW4kgnQIL2LLuEYKs%2BB0bcga1%2FUJlkgPywVN6GQUumeCGVM2aZqY8em2doH%2F6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA%3D%3D@127.0.0.1:6015/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true' },
> headers: 
> { accept: 'application/json',
> host: '127.0.0.1:6015',
> authorization: 'Basic X2hvb2RpZTpDbDBFaEVKbmNodTV1QmFvbUtmTHJYRktUT0hST2xld21wb09aeXROTTNWcWpPUzdHcE9QdmFxMGVqLzFSWmVlaEZaSWFlTVNWUGJRa3l3Ry9oSkx6bHdWeW5vbndxVU1TaEVhanFuY29wZy9uOHlmSXFOME1KY1NZaG4xREFiTFk3dmhhVXpEbWxFLzdoVW1GekRFaCszNkxaRlRQR3JKZ0d6U1VsV2Fub3dHVmhkemFkdU1PZ016aXVLV3R1QzVMZ3QxZVI0eUhjQUp1d3IrYlJ6bjdkVzRrZ25RSUwyTEx1RVlLcytCMGJjZ2ExL1VKbGtnUHl3Vk42R1FVdW1lQ0dWTTJhWnFZOGVtMmRvSC82UWNUdVdKMVp5aGJpR3Z6WGZ4Z3YzaEJncm5IUkJCU2tSNnIzUDFMMlBDbnFkeWVtOW1tMEtEVjF0N2ZPS3JOQ0t1dkE9PQ==' },
> encoding: 'utf-8',
> changes_query: 
> { since: 7,
> feed: 'continuous',
> heartbeat: 30000,
> include_docs: true },
> pool: {},
> dests: [],
> __isRequestRequest: true,
> _redirectsFollowed: 0,
> maxRedirects: 10,
> followRedirect: true,
> followAllRedirects: false,
> redirects: [],
> setHost: true,
> originalCookieHeader: undefined,
> _jar: undefined,
> port: '6015',
> host: '127.0.0.1',
> clientErrorHandler: [Function],
> _parserErrorHandler: [Function],
> path: '/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
> _user: '_hoodie',
> _pass: 'Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej/1RZeehFZIaeMSVPbQkywG/hJLzlwVynonwqUMShEajqncopg/n8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE/7hUmFzDEh+36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr+bRzn7dW4kgnQIL2LLuEYKs+B0bcga1/UJlkgPywVN6GQUumeCGVM2aZqY8em2doH/6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA==',
> _hasAuth: true,
> _sentAuth: true,
> httpModule: 
> { parsers: [Object],
> STATUS_CODES: [Object],
> IncomingMessage: [Object],
> OutgoingMessage: [Object],
> ServerResponse: [Object],
> Agent: [Object],
> globalAgent: [Object],
> ClientRequest: [Object],
> request: [Function],
> get: [Function],
> Server: [Object],
> createServer: [Function],
> _connectionListener: [Function: connectionListener],
> Client: [Function: deprecated],
> createClient: [Function: deprecated] },
> agentClass: { [Function: Agent] super: [Object], defaultMaxSockets: 5 },
> agent: 
> { domain: null,
> _events: [Object],
> _maxListeners: 10,
> options: {},
> requests: {},
> sockets: [Object],
> maxSockets: 5,
> createConnection: [Function] },
> _started: true,
> href: 'http://_hoodie:Cl0EhEJnchu5uBaomKfLrXFKTOHROlewmpoOZytNM3VqjOS7GpOPvaq0ej%2F1RZeehFZIaeMSVPbQkywG%2FhJLzlwVynonwqUMShEajqncopg%2Fn8yfIqN0MJcSYhn1DAbLY7vhaUzDmlE%2F7hUmFzDEh%2B36LZFTPGrJgGzSUlWanowGVhdzaduMOgMziuKWtuC5Lgt1eR4yHcAJuwr%2BbRzn7dW4kgnQIL2LLuEYKs%2BB0bcga1%2FUJlkgPywVN6GQUumeCGVM2aZqY8em2doH%2F6QcTuWJ1ZyhbiGvzXfxgv3hBgrnHRBBSkR6r3P1L2PCnqdyem9mm0KDV1t7fOKrNCKuvA%3D%3D@127.0.0.1:6015/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
> req: 
> { domain: [Object],
> _events: [Object],
> _maxListeners: 10,
> output: [],
> outputEncodings: [],
> writable: true,
> _last: true,
> chunkedEncoding: false,
> shouldKeepAlive: true,
> useChunkedEncodingByDefault: false,
> sendDate: false,
> _headerSent: true,
> _header: 'GET /user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true HTTP/1.1\r\naccept: application/json\r\nhost: 127.0.0.1:6015\r\nauthorization: Basic X2hvb2RpZTpDbDBFaEVKbmNodTV1QmFvbUtmTHJYRktUT0hST2xld21wb09aeXROTTNWcWpPUzdHcE9QdmFxMGVqLzFSWmVlaEZaSWFlTVNWUGJRa3l3Ry9oSkx6bHdWeW5vbndxVU1TaEVhanFuY29wZy9uOHlmSXFOME1KY1NZaG4xREFiTFk3dmhhVXpEbWxFLzdoVW1GekRFaCszNkxaRlRQR3JKZ0d6U1VsV2Fub3dHVmhkemFkdU1PZ016aXVLV3R1QzVMZ3QxZVI0eUhjQUp1d3IrYlJ6bjdkVzRrZ25RSUwyTEx1RVlLcytCMGJjZ2ExL1VKbGtnUHl3Vk42R1FVdW1lQ0dWTTJhWnFZOGVtMmRvSC82UWNUdVdKMVp5aGJpR3Z6WGZ4Z3YzaEJncm5IUkJCU2tSNnIzUDFMMlBDbnFkeWVtOW1tMEtEVjF0N2ZPS3JOQ0t1dkE9PQ==\r\nConnection: keep-alive\r\n\r\n',
> _hasBody: true,
> _trailer: '',
> finished: true,
> _hangupClose: false,
> socket: [Object],
> connection: [Object],
> agent: [Object],
> socketPath: undefined,
> method: 'GET',
> path: '/user%2F789bh5f/_changes?since=7&feed=continuous&heartbeat=30000&include_docs=true',
> _headers: [Object],
> _headerNames: [Object],
> parser: [Object] },
> ntick: true },
> domain: 
> { domain: null,
> _events: { error: [Function] },
> _maxListeners: 10,
> members: [] },
> domainThrown: false }
> 
> events.js:53
> EventEmitter.prototype.emit = function emit(type) {
> 
> —
> Reply to this email directly or view it on GitHub.
> 

-- 
Professional Support for Apache CouchDB:
http://www.neighbourhood.ie/couchdb-support/

**Comment by quaxquax** [comment]
Thanks Jan, for taking a look.

The hoodie-server version of the app that I reported here fist was 3.1.0 (2015-03-01).

Overnight I also started an unaltered 'crashtest' app freshly minted with 'hoodie new' and otherwise unaltered.  It also eventually crashed with just:

events.js:53
EventEmitter.prototype.emit = function emit(type) {                                           ^
RangeError: Maximum call stack size exceeded

No further error trace.  This one reports hoodie-server version 3.1.1 (2015-03-22).
